cmd/minikube/cmd/start.go

```diff
@@ -402,29 +401,31 @@ func provisionWithDriver(cmd *cobra.Command, ds registry.DriverState, existing *
 	}, nil
 }
 
-func virtualBoxMacOS13PlusWarning(driverName string) {
-	if !driver.IsVirtualBox(driverName) || !detect.MacOS13Plus() {
+// warnDriverDeprecated prints a small deprecation warning if the selected driver is deprecated
+func warnDriverDeprecated(driverName string) {
+	if driverName == "" {
 		return
 	}
-	out.WarningT(`Due to changes in macOS 13+ minikube doesn't currently support VirtualBox. You can use alternative drivers such as 'vfkit', 'qemu', or 'docker'.
-    https://minikube.sigs.k8s.io/docs/drivers/vfkit/
-    https://minikube.sigs.k8s.io/docs/drivers/qemu/
-    https://minikube.sigs.k8s.io/docs/drivers/docker/
-    For more details on the issue see: https://github.com/kubernetes/minikube/issues/15274
-`)
-}
 
-// hyperkitDeprecationWarning prints a deprecation warning for the hyperkit driver
-func hyperkitDeprecationWarning(driverName string) {
-	if !driver.IsHyperKit(driverName) {
-		return
+	for _, d := range registry.List() {
+		if d.Priority != registry.Deprecated {
+			continue
+		}
+
+		// match driver name
+		if d.Name == driverName {
+			out.WarningT("The '{{.name}}' driver is deprecated and will be removed in a future release.\n    You can use alternative drivers: {{.drivers}}.", out.V{"name": d.Name, "drivers": strings.Join(driver.SupportedDrivers(), ", ")})
+			return
+		}
+
+		// match any alias
+		for _, a := range d.Alias {
+			if a == driverName {
+				out.WarningT("The '{{.name}}' driver is deprecated and will be removed in a future release.\n    You can use alternative drivers: {{.drivers}}.", out.V{"name": d.Name, "drivers": strings.Join(driver.SupportedDrivers(), ", ")})
+				return
+			}
```

Comment:
It does not make sense to iterate over all drivers and check both driver name and all aliases. The registry keeps a map of drivers by name and alias, so getting a driver should be a single map lookup using the name and if not found a second map lookup using the alias.